src/iceberg/avro/avro_schema_util.cc

```diff
@@ -181,8 +231,10 @@ Status ToAvroNodeVisitor::Visit(const StructType& type, ::avro::NodePtr* node) {
     ::avro::NodePtr field_node;
     ICEBERG_RETURN_UNEXPECTED(Visit(sub_field, &field_node));
 
-    // TODO(gangwu): sanitize field name
-    (*node)->addName(std::string(sub_field.name()));
+    bool isValidFieldName = validAvroName(std::string(sub_field.name()));
```

Comment:
If the name has been sanitized, you need to follow this: https://github.com/apache/iceberg/blob/be03c998d96d0d1fae13aa8c53d6c7c87e2d60ba/core/src/main/java/org/apache/iceberg/avro/TypeToSchema.java#L125-L127